bundles/org.openhab.binding.lghombot/src/main/java/org/openhab/binding/lghombot/internal/LGHomBotHandler.java

```diff
@@ -0,0 +1,722 @@
+            case CHANNEL_NICKNAME:
+                updateState(channelUID, StringType.valueOf(currentNickname));
+                break;
+            case CHANNEL_CAMERA:
+                parseImage();
+                updateState(channelUID, currentImage);
+                break;
+            case CHANNEL_LAST_CLEAN:
+                updateState(channelUID, currentLastClean);
+                break;
+            case CHANNEL_MAP:
+                parseMap();
+                updateState(channelUID, currentMap);
+                break;
+            case CHANNEL_MONDAY:
+            case CHANNEL_TUESDAY:
+            case CHANNEL_WEDNESDAY:
+            case CHANNEL_THURSDAY:
+            case CHANNEL_FRIDAY:
+            case CHANNEL_SATURDAY:
+            case CHANNEL_SUNDAY:
+                refreshCounter = 1;
+                break;
+            default:
+                logger.warn("Channel refresh for {} not implemented!", channelUID.getId());
+        }
+    }
+
+    private void updateAllChannels() {
+        if (disposed) {
+            return;
+        }
+        if (refreshCounter > 0) {
+            refreshCounter--;
+            if (refreshCounter == 0) {
+                fetchSchedule();
+                return;
+            }
+            if (refreshCounter % 5 == 1) {
+                parseImage();
+                updateState(new ChannelUID(getThing().getUID(), CHANNEL_CAMERA), currentImage);
+                return;
+            }
+        }
+        String status = null;
+        String url = buildHttpAddress("/status.txt");
+        try {
+            status = HttpUtil.executeUrl("GET", url, 1000);
+            if (getThing().getStatus() != ThingStatus.ONLINE) {
+                updateStatus(ThingStatus.ONLINE);
+            }
+        } catch (IOException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+        }
+        if (status != null && !status.isEmpty()) {
+            boolean parsingOk = true;
+            String[] rows = status.split("\\r?\\n");
+            ChannelUID channel;
+            for (String row : rows) {
+                int idx = row.indexOf('=');
+                if (idx == -1) {
+                    continue;
+                }
+                final String key = row.substring(0, idx);
+                String value = row.substring(idx + 1).replace("\"", "");
+                switch (key) {
+                    case "JSON_ROBOT_STATE":
+                        if (value.isEmpty()) {
+                            value = HBSTATE_UNKNOWN;
+                        }
+                        if (!value.equals(currentState)) {
+                            currentState = value;
+                            channel = new ChannelUID(getThing().getUID(), CHANNEL_STATE);
+                            updateState(channel, StringType.valueOf(value));
+
+                            if (value.equals(HBSTATE_WORKING) || value.equals(HBSTATE_BACKMOVING)
```

Comment:
Position literals first in `String` comparisons - that way if the `String` is `null` you won't get a `NullPointerException`, it'll just return `false`. [source: PMD](http://pmd.sourceforge.net/pmd-4.3.0/rules/design.html#PositionLiteralsFirstInComparisons)

Reply:
Even though its null checked already, its still best practice.